### PR TITLE
Do Not Track Active Generation in State

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

- Modifies `switch-generation` command to check whether there is an active next generation when attempting to set the generation to "next".
- Removes methods relating to generation switching from API clients and servers.
- Removes the `Active` member from `Generation` in the state package along with logic relating to it.
- Replaces transaction assertions that deal with the `Active` value, with logic to ensure that a generation is not yet completed (which `Active` was an effective proxy for).

## QA steps

- `export JUJU_DEV_FEATURE_FLAGS=generations`.
- Bootstrap.
- `juju deploy redis -n 2`
- `juju add-generation`.
- `juju advance-generation redis/0`.
- `juju cancel-generation` should fail.
- `juju switch-generation current`.
- `juju advance-generation redis` (all remaining will auto-complete generation).
- `juju switch-generation next` should fail due to no "next" generation being found.

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

File: api/modelgeneration/modelgeneration.go [hunk 1]
```diff
@@ -27,9 +26,9 @@ func NewClient(st base.APICallCloser) *Client {
 }
 
 // AddGeneration adds a model generation to the config.
-func (c *Client) AddGeneration(model names.ModelTag) error {
+func (c *Client) AddGeneration(modelUUID string) error {
```
Comment (manadart) [inline api/modelgeneration/modelgeneration.go]: I poked the tag conversion in here for the following reasons:
- It keeps the commands cleaner, as they don't need to do conversion for what is essentially a wire format.
- We only have to import `names` in one file and do all conversion immediately pre-flight.

**Comment by manadart** [comment]
[determine storage mechanism for tracking generations on/off. Current always on, next on/off.](https://trello.com/c/RGYyx3Ul/857-determine-storage-mechanism-for-tracking-generations-on-off-current-always-on-next-on-off)

**Comment by jameinel** [comment]
How much of this is in https://github.com/juju/juju/pull/9729 ?

I'll try to give it a look later, but at least some of the comments seem to overlap.

**Comment by manadart** [comment]
#9729 is based on this, so this one is a proper subset. I will apply your comments there to this PR, and rebase when it can be landed.

**Comment by manadart** [comment]
Actually, the changes are better applied in PR reviewed. I will $$merge$$ this one and refactor the other.

**Pull request merged**